app/components/wallet/PdfTransactionList.js

```diff
@@ -118,14 +119,24 @@ function renderTransactionBody (tx) {
           </Text>
           {out.senders.map(sender => {
             return (
-              <Text style={styles.addresses}>From: {sender}</Text>
+              <Text style={styles.addresses}>From: {sender + getAddressName(sender, account)}</Text>
             )
           })}
-          <Text style={styles.addresses}>To: {out.recipient}</Text>
+          <Text style={styles.addresses}>To: {out.recipient + getAddressName(out.recipient, account)}</Text>
         </View>
       )
     })
   }
 }
 
+function getAddressName (sender, account) {
+  const { address_book: addressbook } = account
+  const { contacts } = addressbook
+  const singleContacts = contacts.filter(contact => !(contact.recipient instanceof Array))
+  const contact = find(singleContacts, contact => contact.recipient === sender)
+  const wallet = account.wallet_for_address(sender)
+
+  return contact && contact.contact_name ? ' (contact: ' + contact.contact_name + ')' : wallet && wallet.wallet_name ? ' (wallet: ' + wallet.wallet_name + ')' : ''
```

Comment:
It might be a sport to fit everything on one line but it really make things very unreadable. Inline if-else cases are cute, but don't over-do it. Please for situations like this use regular if-else statements and don't be afraid of using new lines.